src/oge/data_cleaning.py

```diff
@@ -946,6 +976,64 @@ def assign_fuel_type_to_cems(cems, year, primary_fuel_table):
         filler_column="energy_source_code_epa",
     )
 
+    # as a final backstop, assign an energy_source_code based on the plant's primary
```

Comment:
I'm wondering if this should be added as a separate function for ease of readibility.

Reply:
LMK if you want this to be done in this PR

Reply:
Sure, let's go ahead and do this so it doesn't get lost.

Reply:
Done. LMK what you think. Happy to change function name, etc.